ZUGFeRD/InvoiceDescriptor22UBLWriter.cs

```diff
@@ -439,110 +439,131 @@ public override void Save(InvoiceDescriptor descriptor, Stream stream, ZUGFeRDFo
-                Writer.WriteElementString("cbc", "Name", tradeLineItem.Name);
+            //Writer.WriteElementString("cbc", "InvoicedQuantity", tradeLineItem.BilledQuantity.ToString());
+            Writer.WriteStartElement("cbc", "InvoicedQuantity");
+            Writer.WriteAttributeString("unitCode", tradeLineItem.UnitCode.EnumToString());
+            Writer.WriteValue(_formatDecimal(tradeLineItem.BilledQuantity));
+            Writer.WriteEndElement();
 
-                if (!string.IsNullOrWhiteSpace(tradeLineItem.SellerAssignedID))
-                {
-                    Writer.WriteStartElement("cac", "SellersItemIdentification");
-                    Writer.WriteElementString("cbc", "ID", tradeLineItem.SellerAssignedID);
-                    Writer.WriteEndElement(); //!SellersItemIdentification
-                }
 
-                if (!string.IsNullOrWhiteSpace(tradeLineItem.BuyerAssignedID))
-                {
-                    Writer.WriteStartElement("cac", "BuyersItemIdentification");
-                    Writer.WriteElementString("cbc", "ID", tradeLineItem.BuyerAssignedID);
-                    Writer.WriteEndElement(); //!BuyersItemIdentification
-                }
+            //Writer.WriteElementString("cbc", "LineExtensionAmount", tradeLineItem.LineTotalAmount.ToString());
+            Writer.WriteStartElement("cbc", "LineExtensionAmount");
+            Writer.WriteAttributeString("currencyID", this.Descriptor.Currency.EnumToString());
+            Writer.WriteValue(_formatDecimal(tradeLineItem.LineTotalAmount));
+            Writer.WriteEndElement();
 
-                _writeApplicableProductCharacteristics(Writer, tradeLineItem.ApplicableProductCharacteristics);
-                _writeIncludedReferencedProducts(Writer, tradeLineItem.IncludedReferencedProducts);
-                _WriteCommodityClassification(Writer, tradeLineItem.GetDesignatedProductClassifications());
 
-                //[UBL-SR-48] - Invoice lines shall have one and only one classified tax category.
-                Writer.WriteStartElement("cac", "ClassifiedTaxCategory");
-                Writer.WriteElementString("cbc", "ID", tradeLineItem.TaxCategoryCode.EnumToString());
-                Writer.WriteElementString("cbc", "Percent", _formatDecimal(tradeLineItem.TaxPercent));
+            Writer.WriteStartElement("cac", "Item");
 
-                Writer.WriteStartElement("cac", "TaxScheme");
-                Writer.WriteElementString("cbc", "ID", tradeLineItem.TaxType.EnumToString());
-                Writer.WriteEndElement();// !TaxScheme
+            Writer.WriteOptionalElementString("cbc", "Description", tradeLineItem.Description);
+            Writer.WriteElementString("cbc", "Name", tradeLineItem.Name);
 
-                Writer.WriteEndElement();// !ClassifiedTaxCategory
+            if (!string.IsNullOrWhiteSpace(tradeLineItem.SellerAssignedID))
```

Comment:
I guess you just sorted the elements according to the documentation?

Reply:
alright. Can you please quickly check that the order of elements is still in line with the documentation?
Should just be a matter of seconds.

Quick reference here:
https://docs.peppol.eu/poacc/billing/3.0/syntax/ubl-invoice/

Reply:
Order still fits :)